addons/mail_group/security/mail_group_security.xml

```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo noupdate="1">
+    <record id="mail_group_message_rule_public" model="ir.rule">
+        <field name="name">Mail Group Message: Only accepted message are accessible</field>
+        <field name="model_id" ref="model_mail_group_message"/>
+        <field name="domain_force">[('moderation_status', '=', 'accepted')]</field>
+        <field name="groups" eval="[(4, ref('base.group_public')), (4, ref('base.group_portal'))]"/>
+        <field name="perm_create" eval="True"/>
+        <field name="perm_read" eval="True"/>
+        <field name="perm_write" eval="True"/>
+        <field name="perm_unlink" eval="True"/>
+    </record>
+    <record id="mail_group_message_rule_user" model="ir.rule">
+        <field name="name">Mail Group Message: Non-accepted messages are accessible only by moderators</field>
+        <field name="model_id" ref="model_mail_group_message"/>
+        <field name="domain_force">['|', '|', ('moderation_status', '=', 'accepted'), ('mail_group_id.moderator_ids', 'in', user.id), ('mail_group_id.moderation', '=', False)]</field>
```

Comment:
I can not give the read access to the member because member might not be related to a user (public user) or even might have no partner. Here only moderators have full access on moderated group (to not restrict the moderation to `group_system`)

Reply:
Done with `group_mailing_list_administrator` (see above comment)